indico/modules/events/editing/operations.py

```diff
@@ -269,38 +302,44 @@ def create_new_tag(event, code, title, color, system=False):
     return tag
 
 
-def update_tag(tag, code=None, title=None, color=None):
-    if code:
-        tag.code = code
-    if title:
-        tag.title = title
-    if color:
-        tag.color = color
+def update_tag(tag, updates):
+    changes = tag.populate_from_dict({k: v for k, v in updates.items() if v is not None})
     db.session.flush()
     logger.info('Tag %r updated by %r', tag, session.user)
+    log_fields = {'code': 'Code', 'title': 'Title', 'color': 'Color'}
+    tag.event.log(EventLogRealm.management, LogKind.change, 'Editing', f'Tag {tag.verbose_title} updated',
+                  session.user, data={'Changes': make_diff_log(changes, log_fields)})
 
 
 def delete_tag(tag):
     logger.info('Tag %r deleted by %r', tag, session.user)
     db.session.delete(tag)
+    tag.event.log(EventLogRealm.management, LogKind.negative, 'Editing', f'Tag {tag.verbose_title} deleted',
+                  session.user)
 
 
 def create_new_file_type(event, editable_type, **data):
     file_type = EditingFileType(event=event, type=editable_type)
-    file_type.populate_from_dict(data, keys=FILE_TYPE_ATTRS)
+    file_type.populate_from_dict(data, keys=FILE_TYPE_ATTRS.keys())
```

Comment:
Is this change actually needed? The way `keys` is used, passing a dict should work just fine (`if keys and key not in keys`)